flytekit/core/type_engine.py

```diff
@@ -24,6 +23,7 @@
 from google.protobuf.struct_pb2 import Struct
 from marshmallow_enum import EnumField, LoadDumpOptions
 from mashumaro.mixins.json import DataClassJSONMixin
+from mashumaro.mixins.orjson import DataClassORJSONMixin
```

Comment:
I prefer not to add `orjson` as a hard dependency for `flytekit`. Can you write the PR in way where `orjson` is a soft dependency?

Reply:
Great idea, orjson is a soft dependency now